src/cljs/org/broadinstitute/firecloud_ui/common/table.cljs

```diff
@@ -140,34 +145,52 @@
                    {:backgroundColor (if (even? index) (:background-light style/colors) "#fff")})})
    :get-initial-state
    (fn [{:keys [props]}]
-     (persistence/try-restore
-      {:key (:state-key props)
-       :initial
-       (let [columns (vec (map-indexed (fn [i col]
-                                         {:width (or (:starting-width col) 100)
-                                          :visible? (get col :show-initial? true)
-                                          :index i
-                                          :display-index i})
-                                       (:columns props)))
-             initial-sort-column (or (first (filter #(contains? % :sort-initial)
-                                                    (map merge (:columns props) columns)))
-                                     (when (:always-sort? props)
-                                       (first (map merge (:columns props) columns))))]
-         {:columns columns
-          :dragging? false
-          :filter-group-index (or (:initial-filter-group-index props) 0)
-          :query-params (merge
-                         {:current-page 1 :rows-per-page initial-rows-per-page
-                          :filter-text ""}
-                         (when initial-sort-column
-                           {:sort-column (:index initial-sort-column)
-                            ; default needed when forcing sort
-                            :sort-order (or (:sort-initial initial-sort-column) :asc)}))})}))
+     (let [given-by-header (->> (:columns props)
+                                (map-indexed (fn [index {:keys [header header-key] :as col}]
+                                               [(or header-key header) (assoc col :declared-index index)]))
+                                (into {}))]
+       (merge {:given-columns-by-header given-by-header
+               :dragging? false}
+              (persistence/try-restore
+               {:key (:state-key props)
+                :validator (fn [stored-value] (= (set (keys stored-value)) #{:column-meta :filter-group-index :query-params}))
```

Comment:
I don't _think_ that works here, we need a list of keys and that gets the values.